httpclient/httpclient.go

```diff
@@ -26,3 +31,138 @@ func CloneHTTPHeader(in http.Header) http.Header {
 	}
 	return out
 }
+
+// New wraps delegate transports with logging, rate limiting, retryable, request id
+// and returns an error if any occurs.
+func New(cfg *Config) (*http.Client, error) {
+	var err error
+	var delegate http.RoundTripper
+
+	delegate = http.DefaultTransport.(*http.Transport).Clone()
+
+	if cfg.Log.Enabled {
+		opts := cfg.Log.TransportOpts()
+		delegate = NewLoggingRoundTripperWithOpts(delegate, opts)
+	}
+
+	if cfg.RateLimits.Enabled {
+		delegate, err = NewRateLimitingRoundTripperWithOpts(
+			delegate, cfg.RateLimits.Limit, cfg.RateLimits.TransportOpts(),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("create rate limiting round tripper: %w", err)
+		}
+	}
+
+	delegate = NewRequestIDRoundTripper(delegate)
+
+	if cfg.Retries.Enabled {
+		opts := cfg.Retries.TransportOpts()
+		opts.BackoffPolicy = cfg.Retries.GetPolicy()
+		delegate, err = NewRetryableRoundTripperWithOpts(delegate, opts)
+		if err != nil {
+			return nil, fmt.Errorf("create retryable round tripper: %w", err)
+		}
+	}
+
+	return &http.Client{Transport: delegate, Timeout: cfg.Timeout}, nil
+}
+
+// Must wraps delegate transports with logging, rate limiting, retryable, request id
+// and panics if any error occurs.
+func Must(cfg *Config) *http.Client {
+	client, err := New(cfg)
+	if err != nil {
+		panic(err)
+	}
+
+	return client
+}
+
+// Opts provides options for NewWithOpts and MustWithOpts functions.
+type Opts struct {
+	// UserAgent is a user agent string.
+	UserAgent string
+
+	// ReqType is a type of request. e.g. service 'auth-service', an action 'login' or specific information to correlate.
+	ReqType string
```

Comment:
Since `ReqType` is part of the public interface, let’s use the full form for better readability: `ReqType` -> `RequestType`.